src/edi_energy_scraper/__init__.py

```diff
@@ -300,6 +301,25 @@ async def _download(
             raise
         return file_path
 
+    def get_edifact_format(self, path: Path) -> tuple[EdifactFormatVersion, list[EdifactFormat] | None]:
+        """
+        Determines the edifact format of a given file
+        """
+        filename = path.stem
+        date_string = filename.split("_")[-1]  # Assuming date is in the last part of filename
+        date_format = "%Y%m%d"
+        date = datetime.datetime.strptime(date_string, date_format)
+        date = date.replace(tzinfo=datetime.timezone.utc)
```

Comment:
die annahme, die hier drin steckt ist, dass die `%Y%m%d` date-strings in utc gemeint seien. tatsächlich sind sie ja aber berliner lokalzeit. also wenn auf edi-energy.de steht "gültig ab 03.04.2024", dann ist das ja nicht UTC, sondern MEZ.

das klappt glaube ich nur zufällig, weil `filename.split("_")[-1]` das from-date ist und wir hier 1-2stunden zu weit in der zukunft sind und der deutsche utc-offset immer >0 ist.

in kohlrahbi haben wir schon was sehr ähnliches gebaut:
https://github.com/Hochfrequenz/kohlrahbi/blob/851cdf688697e7e2960fdfe52a19f13d26bfc834/src/kohlrahbi/read_functions.py#L54

daran könntest du dich orientieren.

du kannst es auch so lassen, aber dann schreib bitte dazu, unter welchen annahmen das klappt.

Reply:
https://github.com/Hochfrequenz/edi_energy_scraper/pull/124/commits/b50f7f268a9576abdc3ae9c269733745bcb190c2